### PR TITLE
Exporting TreeSE/MAE section

### PR DESCRIPTION
**Pull request by ElySeraidarian**
Issue #330 - Adding a section to explain how to export TSE/MAE data containers in order to use them in Julia or Python (using feather files). 
Not sure of the positioning of the section & the path for output file.

Note : Indicates 3 files changed but only 1 changed.

### CODE REVIEW

**Comment by ElySeraidarian** [comment]
Build fails because can't find package feather : I don't understand where should I add it (added in DESCRIPTION & csv file)

File: inst/pages/04_containers.qmd [hunk 1]
```diff
@@ -740,3 +740,39 @@ Conversion is possible between other data formats. Interested readers can refer
 * [makeTreeSEFromBiom](https://microbiome.github.io/mia/reference/makeTreeSEFromBiom.html)  
 * [importMetaPhlAn](https://microbiome.github.io/mia/reference/importMetaPhlAn.html)  
 * [importQZA](https://microbiome.github.io/mia/reference/importQIIME2.html)
+
+## Exporting data container {#sec-exporting-data-container}
+
+### Export TreeSummarizedExperiment
+
+Exporting a `TreeSE` data container can be done using *feather* package. TSE object has to be converted into a dataframe (`DataFrame`). The output file is a .feather file, which can be imported in other languages such as Julia or Python. For information, have a look at the [FeatherFile](https://juliapackages.com/p/featherfiles) Julia package and [feather-format](https://pypi.org/project/feather-format/) Python library.
```
Comment (TuomasBorman) [inline inst/pages/04_containers.qmd]: Does also data.frame object work? DataFrame and data.frame are different. DF inherits data.frame. One option is also basic csv file (you could add that).

File: inst/pages/04_containers.qmd [hunk 2]
```diff
@@ -740,3 +740,39 @@ Conversion is possible between other data formats. Interested readers can refer
...
+
+```{r, eval=FALSE}
+# Import TSE
+library(mia)
+data(GlobalPatterns, package="mia")
+tse <- GlobalPatterns
+# Convert into a DataFrame
+tse <- transformAssay(tse, MARGIN = "samples", method="relabundance")
+molten_tse <- mia::meltAssay(tse,
```
Comment (TuomasBorman) [inline inst/pages/04_containers.qmd]: remove mia::

File: inst/pages/04_containers.qmd [hunk 3]
```diff
@@ -740,3 +740,39 @@ Conversion is possible between other data formats. Interested readers can refer
...
+Exporting a `MultiAssayExperiment` data container can also be done using *feather* package.
+
+```{r, eval=FALSE}
+# Convert into a DataFrame
+molten_mae <- longFormat(mae)
+molten_mae <- data.frame(molten_mae)
+path <- "dirname(rstudioapi::getSourceEditorContext()$path)/../../../mae.feather"
+write_feather(molten_mae,path)
+```
+
```
Comment (TuomasBorman) [inline inst/pages/04_containers.qmd:797]: Remove extra line (only one line)

File: inst/pages/04_containers.qmd [hunk 4]
```diff
@@ -740,3 +740,39 @@ Conversion is possible between other data formats. Interested readers can refer
...
+### Export MultiAssayExperiment
+
+Exporting a `MultiAssayExperiment` data container can also be done using *feather* package.
+
+```{r, eval=FALSE}
+# Convert into a DataFrame
+molten_mae <- longFormat(mae)
+molten_mae <- data.frame(molten_mae)
+path <- "dirname(rstudioapi::getSourceEditorContext()$path)/../../../mae.feather"
```
Comment (TuomasBorman) [inline inst/pages/04_containers.qmd]: You could add just arbitrary path. I added eval=FALSE, so that nothing is really saved

Comment (TuomasBorman) [reply inst/pages/04_containers.qmd]: like "path/to/mae.feather"

**Comment by antagomir** [comment]
up

**Comment by ElySeraidarian** [comment]
I have made fixes after review but build fails because package not found. However works locally when building the book (imports the package).

**Comment by antagomir** [comment]
@ake123 might know how to solve?

**Comment by ElySeraidarian** [comment]
One way to solve it would be just removing the package from build as we use eval=FALSE it will not be an issue. However, if the user wants to use the code, he will have to install it manually

**Comment by antagomir** [comment]
Check with @ake123 if there is a real solution, and resolve the remaining comments.

**Comment by ake123** [comment]
I forked his repo and tested it and seems to be working fine

**Comment by antagomir** [comment]
> I forked his repo and tested it and seems to be working fine

Yes but how about the failing builds?

**Comment by ake123** [comment]
It's strange because I tested it here (https://github.com/Akejeb/OMA_ely/actions). seems to be fine for me.

**Comment by antagomir** [comment]
How shall we proceed in order to get this solved - any suggestion?

**Comment by TuomasBorman** [comment]
I think it is not wise to save anything, so we can just put eval=FALSE and show how it can be done. There is probably no informative message about saving part so nothing is lost

Then we can remove the package from dependencies

**Comment by TuomasBorman** [comment]
This issue of package that cannot be found is caused because GHA saves cache. If the version is not bumped, GHA expects that the same cache can be used and no new packages are installed (or that is what I think)

**Comment by TuomasBorman** [comment]
Ohh, eval=FALSE already. SO just remove the package from DESCRIPTION. That is also good in that way that we have less dependencies

**Comment by ElySeraidarian** [comment]
Here we are

**Comment by antagomir** [comment]
merge?

**Pull request merged**